sky/specs/style-guide.md

```diff
@@ -8,44 +7,114 @@ that everyone, whether reading the code for the first time or
 maintaining it for years, can quickly determine what the code does. A
 secondary goal is avoiding arguments when there are disagreements.
 
-All languages
--------------
+## All languages
 
-Avoid checking in commented-out code. It will bitrot too fast to be
-useful, and will confuse people maintaining the code.
+### AVOID checking in commented-out code
 
-Avoid checking in comments that ask questions (like "What should this
-be?"). Find the answers to the questions, or describe the confusion,
-including references to where you found answers. ("According to this
-specification, this should be 2.0, but according to that
-specification, it should be 3.0. We split the difference and went with
-2.5, because we didn't know what else to do.")
+It will bitrot too fast to be useful, and will confuse people maintaining the
+code.
 
 
+### AVOID checking in comments that ask questions
 
+Find the answers to the questions, or describe the confusion, including
+references to where you found answers.
 
-Dart
-----
+If commenting on a workaround due to a bug, also leave a link to the bug and
+a TODO to clean it up when the bug is fixed.
+
+Example:
+
+```dart
+// BAD:
+
+// What should this be?
+
+// This is a workaround.
+
+
+// GOOD:
+
+// According to this specification, this should be 2.0, but according to that
+// specification, it should be 3.0. We split the difference and went with
+// 2.5, because we didn't know what else to do.
+
+// TODO: Converting color to RGB because class Color doesn't support hex yet.
+//       See http://link/to/a/bug/123
+```
+
+## Dart
 
 In general, follow the [Dart style
-guide](https://www.dartlang.org/articles/style-guide/) for Dart code,
-except where that would contradict this page. (`dartfmt` [doesn't yet
-support](https://github.com/dart-lang/dart_style/issues/442) our style
-rules, so avoid using that for now.)
+guide](https://www.dartlang.org/articles/style-guide/) for Dart code, except
+where that would contradict this page.
 
-Always use the Dart Analyzer. Avoid checking in code that increases
-the output of the analyzer unless you've filed a bug with the Dart
-team. (Use `flutter analyze` to run the analyzer on Flutter code.)
 
-Use assert()s liberally to describe the contracts that you expect your
-code to follow.
+### DON'T use dartfmt (yet)
```

Comment:
Promoting this "don't" from a parenthetical to a heading seems a bit drastic. We don't really want to draw attention to the problems of using dartfmt, it isn't super polite to the people maintaining dartfmt.